### ISSUE TITLE
util/log: sentry reporting not working

### ISSUE THREAD

**Issue by mgartner** [cockroachdb/cockroach#73606]
During the release process for v21.2.3, it was discovered that panics are not successfully reported to Sentry. This is blocking the v21.2.3 release.

To reproduce:

#### 1. Download the binary of the candidate SHA (Build ID `v21.2.2-131-g81d49bd3d2`):

On MacOS:

```
curl https://cockroach-builds.s3.amazonaws.com/cockroach-v21.2.2-131-g81d49bd3d2.darwin-10.9-amd64.tgz | tar -xJ
```

On Linux:

```
curl https://cockroach-builds.s3.amazonaws.com/cockroach-v21.2.2-131-g81d49bd3d2.linux-amd64.tgz | tar -xz
```

#### 2. Run the binary and force a panic:

On MacOS:

```
echo "select crdb_internal.force_panic('testing');" | ./cockroach-v21.2.2-131-g81d49bd3d2.darwin-10.9-amd64/cockroach demo --insecure
```

On Linux:

```
echo "select crdb_internal.force_panic('testing');" | ./cockroach-v21.2.2-131-g81d49bd3d2.linux-amd64/cockroach demo --insecure
```

**Comment by mgartner** [comment]
I'm able to reproduce with v21.2.2 as well.

**Comment by mgartner** [comment]
Each panic that I induced logged an event ID. For example, the `08fa254999234c9d894ac669ded32f70` in the log below is the event ID.

```
* ERROR: [n1,client=127.0.0.1:65396,hostnossl,user=root] Queued as error 08fa254999234c9d894ac669ded32f70
```

The event ID is generated by the sentry SDK, and is only produced in our logs after the HTTP payload upload has completed. Therefore, this is not an issue with the cockroach binary. It's either a problem with the forwarding server or Sentry.

I've removed the release blocker label and reassigned to dev-inf.
